api/institutions/authentication.py

```diff
@@ -84,6 +84,7 @@ def authenticate(self, request):
         family_name = provider['user'].get('familyName')
         middle_names = provider['user'].get('middleNames')
         suffix = provider['user'].get('suffix')
+        department = provider['user'].get('department')
```

Comment:
There is a chance that CAS receives multiple departments and I think OSF should be the one who decides which one to use and makes sure that the one is used consistently. I would suggest OSF try to look for both. If `department` is available, we ignore the `departments`. If not, we take a look at the `departments` and choose one (in a consistent way). Whether a string separated by a delimiter or a list, it's your choice.

```python
department = provider['user'].get('department')
departments = provider['user'].get('departments')
```

Reply:
I thought we decided that CAS would handle any variations on potential departments, because that's the point at which we normalize the rest of the login relationship with the institution. OSF won't have a good way to determine what's right, so it would be an arbitrary choice under most circumstances. About the only thing we could do is to check to see if the same entry that we filled in before is available again, but I suspect that would be pretty well handled by sorting on the CAS side.

Reply:
You are right. It's more sound if CAS sort it. OSF only needs one department and CAS should provide that.